src/jesse_validator_draft4.erl

```diff
@@ -1132,21 +1136,26 @@ check_all_of_(Value, [Schema | Schemas], State) ->
 %%
 %% @private
 check_any_of(Value, [_ | _] = Schemas, State) ->
-  check_any_of_(Value, Schemas, State);
+  check_any_of_(Value, Schemas, State, []);
 check_any_of(_Value, _InvalidSchemas, State) ->
   handle_schema_invalid(?wrong_any_of_schema_array, State).
 
-check_any_of_(Value, [], State) ->
+check_any_of_(Value, [], State, []) ->
   handle_data_invalid(?any_schemas_not_valid, Value, State);
-check_any_of_(Value, [Schema | Schemas], State) ->
+check_any_of_(Value, [], State, SchemaErrors) ->
+  [Errors | _] = lists:sort(fun by_length/2, SchemaErrors),
+  handle_data_invalid({?any_schemas_not_valid, Errors}, Value, State);
+check_any_of_(Value, [Schema | Schemas], State, SchemaErrors) ->
   NumErrsBefore = length(jesse_state:get_error_list(State)),
   case validate_schema(Value, Schema, State) of
     {true, NewState} ->
-        case length(jesse_state:get_error_list(NewState)) of
-            NumErrsBefore -> NewState;
-            _  -> check_any_of_(Value, Schemas, State)
-        end;
-    {false, _} -> check_any_of_(Value, Schemas, State)
+      Errors = jesse_state:get_error_list(NewState),
+      case length(Errors) of
+        NumErrsBefore -> NewState;
+        _  -> check_any_of_(Value, Schemas, State, [Errors | SchemaErrors])
```

Comment:
I'm adding `Errors` in the list of `SchemaErrors` as returned from `jesse_state:get_error_list/1`. Not sure if I should extract the diff from the before and after list of errors returned by `jesse_state:get_error_list/1`.

Reply:
Another option is to always keep only the smallest error: do smth like `check_any_of_(.., shortest_error(PreviousShortestError, NewErrors))`?